### PR TITLE
Add the "contour.linewidths" configuration option

### PR DESCRIPTION
**Pull request by stefraynaud**
## PR Summary

Contours plots are often more beautiful when line widths are smaller
than line widths that are generally used for curves.
This is especially true when ``contour`` is used with ``contourf``, or when
the number of levels if high.

With this PR, one can change default value for contour linewidths independently from normal line widths. The new options is ``"contour.linewidths"``, which defaults to ``None``.  When value is ``None``, it falls back to  ``"lines.linewidth"`` like before.

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [x] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [x] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/contour.py [hunk 1]
```diff
@@ -1749,7 +1752,7 @@ def _initialize_x_y(self, z):
             however introduce rendering artifacts at chunk boundaries depending
             on the backend, the *antialiased* flag and value of *alpha*.
 
-        linewidths : float or sequence of float, default: :rc:`lines.linewidth`
+        linewidths : float or sequence of float
```
Comment (timhoffm) [inline lib/matplotlib/contour.py:1755]: I'd leave the default:
```suggestion
        linewidths : float or sequence of float, default: rc:`contour.linewidth`
```

File: lib/matplotlib/contour.py [hunk 2]
```diff
@@ -1759,6 +1762,9 @@ def _initialize_x_y(self, z):
             If a sequence, the levels in ascending order will be plotted with
             the linewidths in the order specified.
 
+            Defaults to :rc:`contour.linewidths`. If this value is None,
+            it falls back to :rc:`lines.linewidth`.
```
Comment (timhoffm) [inline lib/matplotlib/contour.py:1766]:
```suggestion
            If None, this falls back to :rc:`lines.linewidth`.
```

File: lib/matplotlib/tests/test_contour.py [hunk 3]
```diff
@@ -375,3 +375,17 @@ def test_contour_uneven():
...
     cs = ax.contourf(z, levels=[2, 4, 6, 10, 20])
     fig.colorbar(cs, ax=ax, spacing='uniform')
+
+
+@pytest.mark.parametrize("rcll, rccl, fcl, expected", [
+    (1.23, None, None, 1.23),
+    (1.23, 4.24, None, 4.24),
+    (1.23, 4.24, 5.02, 5.02)
+])
+def test_contour_linewidths(rcll, rccl, fcl, expected):
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_contour.py:385]: Please use more descriptive parameter names. Finding out what these mean is unnecessarily difficult.

**Comment by stefraynaud** [comment]
Linting test is now is failing because of the "default: :rc..." which makes the line too long.

File: lib/matplotlib/contour.py [hunk 4]
```diff
@@ -1749,7 +1752,7 @@ def _initialize_x_y(self, z):
             however introduce rendering artifacts at chunk boundaries depending
             on the backend, the *antialiased* flag and value of *alpha*.
 
-        linewidths : float or sequence of float, default: :rc:`lines.linewidth`
+        linewidths : float or sequence of float, default: :rc:`contour.linewidths`
```
Comment (timhoffm) [inline lib/matplotlib/contour.py:1755]: You can use
```suggestion
        linewidths : float or array-like, default: :rc:`contour.linewidths`
```

which is also more in line with our current documentation standard (https://matplotlib.org/devdocs/devel/documenting_mpl.html#parameter-type-descriptions).

Comment (stefraynaud) [reply lib/matplotlib/contour.py:1755]: Excellent! Thanks.

**Comment by QuLogic** [comment]
Is it plural `linewidths`, or singular `linewidth`? The what's new entry uses both.

**Comment by stefraynaud** [comment]
@QuLogic It's the plural like the argument to ``contour()``. So I made a mistake in the what new entry.

**Comment by QuLogic** [comment]
OK, it only accepts a single float, though? So it should not be plural.

**Comment by timhoffm** [comment]
I agree with @QuLogic 

While the argument is `linewidths` (think: the linewidths of the contours, which can be a list), the rcParam should indeed be `contour.linewidth` (think: the default linewidth of the contour).

**Comment by stefraynaud** [comment]
I agree with you both. My concern was about not adding confusion in the user's mind, despite it's less logical to name it linewidths.
So I renamed it.

**Pull request merged**